homeassistant/components/tolosauna/climate.py

```diff
@@ -0,0 +1,168 @@
+)
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_DRY,
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    CURRENT_HVAC_OFF,
+    FAN_OFF,
+    FAN_ON,
+    HVAC_MODE_DRY,
+    SUPPORT_FAN_MODE,
+    SUPPORT_TARGET_HUMIDITY,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, PRECISION_WHOLE, TEMP_CELSIUS
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ToloSaunaCoordinatorEntity, ToloSaunaUpdateCoordinator
+from .const import (
+    DEFAULT_MAX_HUMIDITY,
+    DEFAULT_MAX_TEMP,
+    DEFAULT_MIN_HUMIDITY,
+    DEFAULT_MIN_TEMP,
+    DOMAIN,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up climate controls for TOLO Sauna."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]["coordinator"]
+    await coordinator.async_config_entry_first_refresh()
+    async_add_entities([SaunaClimate(coordinator, entry)])
+
+
+class SaunaClimate(ToloSaunaCoordinatorEntity, ClimateEntity):
+    """Sauna climate control."""
+
+    def __init__(
+        self, coordinator: ToloSaunaUpdateCoordinator, entry: ConfigEntry
+    ) -> None:
+        """Initialize TOLO Sauna Climate entity."""
+        super().__init__(coordinator, entry)
+
+        self._attr_fan_modes = [FAN_ON, FAN_OFF]
+        self._attr_hvac_modes = [HVAC_MODE_OFF, HVAC_MODE_HEAT, HVAC_MODE_DRY]
+        self._attr_max_humidity = DEFAULT_MAX_HUMIDITY
+        self._attr_max_temp = DEFAULT_MAX_TEMP
+        self._attr_min_humidity = DEFAULT_MIN_HUMIDITY
+        self._attr_min_temp = DEFAULT_MIN_TEMP
+        self._attr_name = "Sauna Climate"
+        self._attr_precision = PRECISION_WHOLE
+        self._attr_supported_features = (
+            SUPPORT_TARGET_TEMPERATURE | SUPPORT_TARGET_HUMIDITY | SUPPORT_FAN_MODE
+        )
+        self._attr_target_temperature_step = 1
+        self._attr_temperature_unit = TEMP_CELSIUS
+        self._attr_unique_id = f"{self._config_entry.entry_id}_climate"
+
+    @property
+    def current_temperature(self) -> int:
+        """Return current temperature."""
+        return self.status.current_temperature
+
+    @property
+    def current_humidity(self) -> int:
+        """Return current humidity."""
+        return self.status.current_humidity
+
+    @property
+    def target_temperature(self) -> int:
+        """Return target temperature."""
+        return self.settings.target_temperature
+
+    @property
+    def target_humidity(self) -> int:
+        """Return target humidity."""
+        return self.settings.target_humidity
+
+    @property
+    def hvac_mode(self) -> str:
+        """Get current HVAC mode."""
+        if self.status.power_on:
+            return HVAC_MODE_HEAT
+        elif not self.status.power_on and self.status.fan_on:
```

Comment:
```suggestion
        if self.status.fan_on:
```
`elif` is not needed after `return`. The check for `not power_on` is also redundant then. So is the `else` at Line 101 (can't figure out how to do multiline suggestions from a phone 😂)